### PR TITLE
Remove plaintext API key from benchmark utility

### PR DESCRIPTION
**Pull request by jpy-git**
### Brief summary of the change made

Spotted a plaintext API key in the `benchmark` function in `util.py`. This is used to post benchmark results and is only executed within a CI context.
Solution is simple, use github secrets. An admin of the repo just needs to add this api key as a repository secret, called `SQLFLUFF_BENCHMARK_API_KEY`, in the settings. This is then picked up by Github actions and supplied to the relevant tox job as an environment variable and can be used within the benchmark function.
image

### Are there any other side effects of this change that we should be aware of?
No, but might be wise to rotate this once you've moved to the repository secrets so people can't mess up your benchmark data/cost you money by posting lots to the API.

### Pull Request checklist
- [X] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `python test/generate_parse_fixture_yml.py` or by running `tox` locally).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

**Comment by jpy-git** [comment]
N.B. I'm expecting some CI tests to fail for this PR as the `SQLFLUFF_BENCHMARK_API_KEY` repository secret needs to  be created first. Then it should start working as expected 👍

File: .github/workflows/ci-tests.yml [hunk 1]
```diff
@@ -85,6 +85,8 @@ jobs:
       run: |
         echo "TOXENV=${{ matrix.jobs }}" >> $GITHUB_ENV
     - name: Run the tests
+      env:
+        SQLFLUFF_BENCHMARK_API_KEY: {{ secrets.SQLFLUFF_BENCHMARK_API_KEY }}
```
Comment (alanmcruickshank) [inline .github/workflows/ci-tests.yml:89]: This line is stopping the CI checks. Does it need a `$` before the `{{` as per the syntax here? https://docs.github.com/en/actions/learn-github-actions/contexts#env-context

Comment (jpy-git) [reply .github/workflows/ci-tests.yml]: @alanmcruickshank you're correct, I've made the update now

Comment (jpy-git) [reply .github/workflows/ci-tests.yml]: Checks look good

**Comment by jpy-git** [comment]
@alanmcruickshank this should be good to merge now?

**Review by alanmcruickshank — APPROVED** [review]
LGTM! Thanks for putting this together

**Comment by jpy-git** [comment]
@alanmcruickshank Fab, I don't have write permissions so you'll have to do the merge for me 😄

**Pull request merged**